.github/workflows/analyze-pr-changes.yml

```diff
@@ -0,0 +1,168 @@
+
+jobs:
+  determine-changes:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Build Docker Image
+        run: docker build coverage --tag opensearch-with-api-plugin
+
+      - name: Run OpenSearch
+        shell: bash -eo pipefail {0}
+        run: |
+          docker run \
+            --name opensearch \
+            --rm -d \
+            -p 9200:9200 -p 9600:9600 \
+            -e "discovery.type=single-node" \
+            -e OPENSEARCH_INITIAL_ADMIN_PASSWORD="$OPENSEARCH_INITIAL_ADMIN_PASSWORD" \
+            opensearch-with-api-plugin
+          
+          for attempt in {1..20}; do
+            sleep 5
+            if curl -ksS -u "admin:$OPENSEARCH_INITIAL_ADMIN_PASSWORD" https://localhost:9200; then
+              echo '=> ready'
+              exit 0
+            fi
+            echo '=> waiting...'
+          done
+          exit 1
+
+      - name: Dump Cluster's API
+        shell: bash -eo pipefail {0}
+        run: curl -ksS -u "admin:$OPENSEARCH_INITIAL_ADMIN_PASSWORD" https://localhost:9200/_plugins/api | jq > /tmp/opensearch-openapi-CLUSTER.json
+
+      - name: Determine Branch Point
+        shell: bash -eo pipefail {0}
+        run: |
+          { 
+            echo "BRANCH_POINT_SHA=$(git merge-base "$BASE_SHA" "$HEAD_SHA")"
+            echo "HEAD_SHA=${HEAD_SHA}"
+          } | tee "$GITHUB_ENV"
+        env:
+          HEAD_SHA: ${{ github.event.pull_request.head.sha }}
+          BASE_SHA: ${{ github.event.pull_request.base.sha }}
+
+      - name: Checkout Original Spec
+        shell: bash -eo pipefail {0}
+        run: git checkout $BRANCH_POINT_SHA
+
+      - name: Build Original Spec
+        shell: bash -eo pipefail {0}
+        run: |
+          npm install
+          npm run merge -- --source ./spec --output /tmp/opensearch-openapi-ORIGINAL.yaml
+
+      - name: Checkout Changed Spec
+        shell: bash -eo pipefail {0}
+        run: git checkout $HEAD_SHA
+
+      - name: Build Changed Spec
+        shell: bash -eo pipefail {0}
+        run: |
+          npm install
+          npm run merge -- --source ./spec --output /tmp/opensearch-openapi-CHANGED.yaml
+
+      - name: Calculate Coverage
+        shell: bash -eo pipefail {0}
+        run: |
+          for variant in ORIGINAL CHANGED ; do
+            echo "Calculating coverage of ${variant}"
+            npm run coverage:spec -- \
+              --cluster /tmp/opensearch-openapi-CLUSTER.json \
+              --specification /tmp/opensearch-openapi-${variant}.yaml \
+              --output /tmp/coverage-api-${variant}.json
+          done
+
+      - name: Upload Coverage Data
+        id: upload-coverage
+        uses: actions/upload-artifact@v4
+        with:
+          name: coverage-api
+          path: |
+            /tmp/coverage-api-ORIGINAL.json
+            /tmp/coverage-api-CHANGED.json
```

Comment:
You could use SHAs instead of ORIGINAL and CHANGED to avoid confusion, running over old data, etc.